### PR TITLE
set cohort download as a background job

### PR DESCRIPTION
**Pull request by dtgreiner**
[//]: # 'remove this if PR is for a release-* branch'
# _Please squash merge this PR_

## Description

Update the cohort download to be performed as a background job

## Type of change
[//]: # 'remove options that are not relevant'
- [X] New feature (adds functionality)

## Checklist before requesting review
- [X] I have performed a self-review of my code
- [X] I have run the code that is being changed under ideal conditions, and it doesn't fail
- [X] My code includes comments and/or descriptive variable names to help other engineers understand the intent (or not applicable)
- [X] My code follows the style guidelines of this project (rubocop)
- [X] I have updated the documentation (or not applicable)
- [ ] If it's not obvious how to test this change, I have provided testing instructions in this PR or the related issue

### CODE REVIEW

**Review by eanders — CHANGES_REQUESTED** [review]
@dtgreiner this looks great overall.  One small change though, we need to adjust the "query_string" saved in the `document_exports` table such that if you were to change the visible columns it should re-generate the download the next time you request it.  

Because of the long expiration in `DocumentExportBehavior` you'll get the same file for 30 days regardless of cohort configuration or who is on the cohort.

Bonus points for adding a client_count=xxx to the `query_string` (which is really a cache key) so that if someone is added or removed, the download would change.

**Comment by dtgreiner** [comment]
@eanders I just pushed an update to include the visible columns and client count in the export query string

File: app/controllers/cohorts_controller.rb [hunk 1]
```diff
@@ -110,6 +111,11 @@ def show
           end
           options
         end
+
+        # included so the excel download can be regenerated when the visible columns change
+        params['visible_columns'] ||= @visible_columns.map { |m| m.column.to_s }.join(',')
+        # included so the excel download can be regenerated if the total number of clients changes
+        params['client_count'] ||= @cohort.search_clients(population: params[:population], user: current_user).count
```
Comment (eanders) [inline app/controllers/cohorts_controller.rb]: Maybe a simple MD5 to keep these small, what do you think?  The idea is simply to force a rebuild if anything has changed.
I'll test:
```suggestion
        # included so the excel download can be regenerated if the total number of clients changes
        params['cache_key'] ||= Digest::MD5.hexdigest(@visible_columns.map { |m| m.column.to_s }.join(',') << @cohort.search_clients(population: params[:population], user: current_user).count)
```

Comment (dtgreiner) [reply app/controllers/cohorts_controller.rb]: This makes sense to me and looks much nicer in the database. It's also working in my tests.

**Pull request merged**